File: Framework/LanguageSupport/python/MMIPython/src/MMIPython/adapter/session/session_content.py
## SPDX-License-Identifier: MIT
## The content of this file has been developed in the context of the MOSIM research project.
## Original author(s): Jannes Lehwald, Janis Sprenger


# -*- coding: utf-8 -*-
"""

"""

# from MMIPython.adapter.scene.mmi_scene import mmi_scene
from MMIPython.core.services import service_access
from MMIPython.core.services.service_access import ServiceAccess


class SessionContent(object):
    
    """
    The session content for a specified session ID.
    
    Arguments
    -----------
    _sessionID : str
        The assigned session ID
    scene_access : SceneAccess
        The scene access of this session
    avatar_content : dict
        Dictionary to for the avatar content
    last_access : datetime
        The last time the session has been used
    """
    
    def __init__(self, sessionID, r_Address):
        """
        Initializes the session content for a given session ID
        
        Parameters
        ----------
        sessionID : str
            The session ID as string
        """
        
        assert (sessionID is not None), "sessionID is invalid."
        assert (isinstance(sessionID, str)), "sessionID is no string."
        
        self._sessionID = sessionID
        self.scene_access = None
        self.service_access = ServiceAccess(r_Address, sessionID)
        self.skeleton_access = None
        self.avatar_content = dict()
        self.last_access = None
        self.mmus = dict()

File: Framework/LanguageSupport/cpp/MMICPP/Adapter/SessionContent.cpp
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Andreas Kaiser, Niclas Delfs, Stephan Adam

#include "SessionContent.h"
#include "SessionData.h"
#include "Adapter/SessionTools.h"


SessionContent::SessionContent(string sessionId) :sessionID{ sessionId },lastAccess{0}
{
	this->serviceAccess = make_unique<ServiceAccess>(SessionData::GetRegisterAddress(),sessionID);
	this->sceneBuffer = make_unique<MMIScene>();
}

ServiceAccess  & SessionContent::GetServiceAccess() const
{
	return *(this->serviceAccess);
}

MMIScene & SessionContent::GetScene() const
{
	return *(this->sceneBuffer);
}

const AvatarContent & SessionContent::GetAvatarContentBySessionID(string sessionID) const
{
	return this->GetAvatarContentByAvatarID(SessionTools::GetSplittedIds(sessionID)[1]);
}

const AvatarContent & MMIStandard::SessionContent::GetAvatarContentByAvatarID(string avatarID) const
{
	auto avatarContentIt = avatarContent.find(avatarID);
	if (avatarContentIt != avatarContent.end()) //avatar content  available
	{
		return *avatarContentIt->second;
	}
	else
	{
		throw runtime_error("Can not find avatar content with id: " + avatarID);
	}
}




